req2flatpak.py

```diff
@@ -695,6 +695,15 @@ def cli_parser() -> argparse.ArgumentParser:
         default=False,
         help="Prints installed packages in requirements.txt format.",
     )
+    try:
+        # get the version number from pyproject.toml
+        __version__ = pkg_resources.get_distribution("req2flatpak").version
+    except pkg_resources.DistributionNotFound:
+        # ... which only works once installed
+        __version__ = "<version unavailable>"
+    parser.add_argument(
+        "--version", action="version", version="%(prog)s " + __version__
+    )
```

Comment:
I tried for a "single source of truth" but a downside of this is that if you copy the file rather than installing, you get:
```
./req2flatpak.py --version
req2flatpak.py <version unavailable>
```

And after I did this I read in the philosophy of the docs:

> This allows req2flatpak to be installed by copy-pasting this file.

Reply:
So I think it would be better to just hardcode the version into the script:
```
__version__ = "0.1.1.dev"
```

Maybe a CI job could ensure the pyproject.toml and this match?

Reply:
Now it is WET but at least the script can be copied.

Reply:
> Maybe a CI job could ensure the pyproject.toml and this match? 

That, plus a pre-commit hook maybe.

Reply:
Where do you have the `.dev` notation from?

Reply:
I had to check, but I think its from https://peps.python.org/pep-0440/

https://peps.python.org/pep-0440/#developmental-releases

Reply:
As for `1.2.3.dev` being the work *leading up to* `1.2.3`, I think that is a consequence of either https://semver.org/ or PEP 440 or maybe both.

Found it:
```python
from packaging.version import Version
Version("1.2.3.dev") < Version("1.2.3")
True
```

Reply:
Interesting. I don't think this conforms to semver2 that's why I was asking.

Reply:
I remember it being confusing!  Particularly `1.2.3-dev` versus `1.2.3.dev` and why some things convert them from `.dev` to `.dev0`.

  * This is what I do in another Python project (not a great argument, I know!)
  * SymPy uses "1.13.dev` in the lead-up to 1.13 (although they may not be on semver precisely).

I really like the idea of bumping the version after release and having some indication of "dev", but I don't really care about the details.

Reply:
version id | conformity
-----------|-----------
`1.2.3-dev`| semver 2.0.0
`1.2.3.dev0`| PEP 440
`1.2.3.dev4`| PEP 440
`1.2.3.dev`| none

Reply:
All look good except the one I proposed in the last row ;-)

Preferences?